### DISCUSSION TITLE
mkdir: cannot create directory '/gpfs/software/genomics/nextflow/nextflow-pipelines/framework': permission denied

### DISCUSSION THREAD

**Discussion by Jega09** [PGScatalog/pgsc_calc#354] [Q&A] [answered]
I am trying to run pgsc on a secured server, specifically Inuvika OVD. I have gotten my PGS score files and sample sheet in the right directory. 

The code below: 
nextflow run pgscatalog/pgscalc -profile conda --input samplesheet.csv --scorefile PGS004887_hmPOS_GRCh38.txt

mkdir: cannot create directory '/gpfs/software/genomics/nextflow/nextflow-pipelines/framework': permission denied

Kindly guide on ways to troubleshoot this.

**Comment by smlmbrt** [comment, marked as answer]
Hi @Jega09, it seems like this is a permissions error that's blocking nextflow from running. I confirm your nextflow runs correctly (https://www.nextflow.io/docs/latest/install.html#install-nextflow), and then try our test command (`nextflow run pgscatalog/pgsc_calc -profile test,conda`) before trying your own data. 

You can also find the specific guide for offline running of the pipeline here: https://pgsc-calc.readthedocs.io/en/latest/how-to/offline.html

**Comment by nebfield** [comment]
By default Nextflow will try to download a copy of our code to:

```
/gpfs/software/genomics/nextflow/nextflow-pipelines/framework
```

If you don't have permissions to edit this directory you'll get an error. 

You can change the location where Nextflow downloads code to by changing the [environment variable `NXF_ASSETS`](https://www.nextflow.io/docs/latest/config.html#environment-variables), e.g.:

```
$ export NXF_ASSETS=$HOME/nxf
$ nextflow run pgscatalog/pgscalc -profile conda --input samplesheet.csv --scorefile PGS004887_hmPOS_GRCh38.txt
```

Or by manually downloading a copy of our code:

```
$ git clone https://github.com/PGScatalog/pgsc_calc.git
$ cd pgsc_calc
$ nextflow run main.nf -profile conda --input samplesheet.csv --scorefile PGS004887_hmPOS_GRCh38.txt
```

(*when running the code locally you need to run main.nf and not pgscatalog/pgsc_calc*)
